graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rest/SimulateRuleRequest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.pipelineprocessor.rest;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+@JsonAutoDetect
+public abstract class SimulateRuleRequest {
+    @JsonProperty
+    public abstract String message();
+
+    @JsonProperty
+    public abstract RuleSource ruleSource();
+
+    @JsonCreator
+    public static SimulateRuleRequest create (@JsonProperty("message") String message,
+                                              @JsonProperty("ruleSource") RuleSource ruleSource) {
```

Comment:
Our API convention is to use snake case instead of camelcase.
```suggestion
                                              @JsonProperty("rule_source") RuleSource ruleSource) {
```

Reply:
fixed